js/ui/components/clubhouse.js

```diff
@@ -289,6 +289,34 @@ var ClubhouseNotificationBanner = new Lang.Class({
                 button.destroy();
         });
     },
+
+    _slideOut: function() {
+        let monitor = Main.layoutManager.primaryMonitor;
+        if (!monitor)
+            return;
+
+        let endX = monitor.x + monitor.width;
+
+        Tweener.addTween(this.actor,
+                         { x: endX,
+                           time: CLUBHOUSE_BANNER_ANIMATION_TIME,
+                           transition: 'easeOutQuad',
+                           onComplete: () => {
+                               this.actor.destroy();
+                               this.actor = null;
+                           }
+                         });
+    },
+
+    destroy: function(shouldSlideOut) {
```

Comment:
I disagree with the semantics of this function. After calling `destroy()`, it is expected that the object is destroyed and thus, any further access to it is invalid. By overriding it, it's effectively breaking this expectation.

I'd suggest, then, renaming `_slideOut()` to `slideOutAndDestroy()`, and removing the `shouldSlideOut` argument from this function.

Reply:
I disagree:
1) This is not a widget, calling it `destroy` has no special meaning as in the behavior you mentioned (which is for widgets only); it does call destroy on its actor but after that it won't access it (even sets it as `null`);
2) Using `slideOutAndDestroy` doesn't do the same, as there's a reason why I use the `shouldSlideOut` arg (sometimes it's supposed to slide-out, others it isn't). And branching this from the caller makes the code less readable.

I hope you understand my point.

Reply:
That's a better explanation. I will update it. Thanks.